### PR TITLE
Prepro auto jason

### PR DESCRIPTION
**Pull request by jasonmotha**
This merge is for the pre-pro refactor branch. 

The commented code is the install specific parts that relate to VM generation. Aka leaving them commented for now so that we can make sure that there is replacement equivalent code before removing. 

https://wiki.canterbury.ac.nz/display/QuakeCore/Plan+for+automated+pre-processing

### CODE REVIEW

File: scripts/submit_hf.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def gen_command_template(params, machine, seed=const.HF_DEFAULT_SEED):
         "fd_statlist": params.FD_STATLIST,
         "hf_bin_path": sim_struct.get_hf_bin_path(params.sim_dir),
         HF_VEL_MOD_1D: params["hf"][HF_VEL_MOD_1D],
-        "duration": params.sim_duration,
```
Comment (jpa198) [inline scripts/submit_hf.py:37]: Is there a scenario where we want sim_duration=0?
Seems like this should be changed to normal dictionary syntax if anything

Comment (jpa198) [reply scripts/submit_hf.py:37]: X( Of course this went on the remove diff, not the add diff

File: templates/gmsim/20.4.1.4/root_defaults.yaml [hunk 2]
```diff
@@ -25,6 +25,11 @@ ims:
   pSA_periods: [0.01, 0.02, 0.03, 0.04, 0.05, 0.075, 0.1, 0.12, 0.15, 0.17, 0.2, 0.25, 0.3, 0.4, 0.5, 0.6, 0.7, 0.75, 0.8, 0.9, 1.0, 1.25, 1.5, 2.0, 2.5, 3.0, 4.0, 5.0, 6.0, 7.5, 10.0]
 flo: 0.25
 dt: 0.02
+VM:
+  hh: 0.4
+  VM_Version: 2.06
+  VM_Topo: BULLDOZED
+  PGV_Threshold: 2
```
Comment (jpa198) [inline templates/gmsim/20.4.1.4/root_defaults.yaml:32]: This is good, I like this

File: scripts/nesi_scripts/vm_params_gen.sl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+fi
+timestamp=`date +%Y%m%d_%H%M%S`
+python $gmsim/workflow/scripts/cybershake/add_to_mgmt_queue.py $MGMT_DB_LOC/mgmt_db_queue $REL_NAME VM_PARAMS running $SLURM_JOB_ID
+
+runtime_fmt="%Y-%m-%d_%H:%M:%S"
+start_time=`date +$runtime_fmt`
+echo $start_time
+
+echo "python $gmsim/Pre-processing/VM/rel2vm_params.py --hh $HH --vm-version $VM_VERSION --vm-topo  $VM_TOPO --pgv 2.0 --ds-multiplier 0.75 $REL_CSV"
+python $gmsim/Pre-processing/VM/rel2vm_params.py --hh $HH --vm-version $VM_VERSION --vm-topo  $VM_TOPO --pgv 2.0 --ds-multiplier 0.75 $REL_CSV
```
Comment (jpa198) [inline scripts/nesi_scripts/vm_params_gen.sl:41]: Having pgv 2.0/ds mult 0.75 is ok for now, but does need to be able to take the validation parameters later

Comment (jasonmotha) [reply scripts/nesi_scripts/vm_params_gen.sl:41]: Agreed. They should be part of the VM parameter section.

File: templates/gmsim/20.4.1.4/root_defaults.yaml (hunk 2 above)
Comment (jasonmotha) [reply templates/gmsim/20.4.1.4/root_defaults.yaml:32]: :smile:

**Comment by jasonmotha** [comment]
Hmm, I need to be carefuller with merging master into my branch.

**Comment by jpa198** [comment]
> Hmm, I need to be carefuller with merging master into my branch.

Eww, that's not a very nice PR any more. Should have done a separate one to merge master in...

**Review by shot0829 — APPROVED** [review]

**Comment by shot0829** [comment]
the longer your feature branch stayed away from the master branch, the more troublesome it will be when you want to merge it eventually. merging the adv_im branch was a pain, insane amount of conflicts.

**Comment by jpa198** [comment]
> the longer your feature branch stayed away from the master branch, the more troublesome it will be when you want to merge it eventually. merging the adv_im branch was a pain, insane amount of conflicts and test failures

Yeah, but if you merge master into the feature branch every week or so then it stays up to date, and can be merged in easily

**Pull request merged**